### PR TITLE
Animates the marbles according to the given move

### PR DESCRIPTION
**Pull request by arimet**
- Use of the Animated API in native react
- Creation of a TranslateView component to allow X or Y movement
- Management of the rendering when the current player plays his move
- Management of the rendering when the opponent player has made a move.

https://user-images.githubusercontent.com/18294269/135275864-9c4f1560-8fa0-4a27-8ce2-be9a48a87b43.mp4

### CODE REVIEW

File: src/GameState/TranslateView.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  let styleView = style;
+  Animated.timing(translateAnimation, {
+    toValue: activated ? actionByDirection.value : 0,
+    duration: 300,
+    useNativeDriver: true,
+  }).start();
+
+  if (direction) {
+    styleView = {
```
Comment (mchaffotte) [inline src/GameState/TranslateView.tsx:22]: Why do you instantiate twice styleView when direction is available?

File: src/GameState/getMarblesCoordinateToAnimate.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  direction: string,
+) => {
+  if (!marbleCoordinate || !graph || !direction) {
+    return [];
+  }
+
+  let currentNode = graph.nodes[`${marbleCoordinate.x},${marbleCoordinate.y}`];
+
+  const nodes = [currentNode];
+  while (true) {
```
Comment (mchaffotte) [inline src/GameState/getMarblesCoordinateToAnimate.tsx]: It is easier to read here when the loop is over. With this, you have to read the whole content of the loop to know when to leave it. So change your loop construction to have a real breaking condition.

File: src/GameState/index.tsx [hunk 3]
```diff
@@ -43,6 +46,28 @@ export default function GameState({ navigation, route, player }: any) {
...
+
+  useEffect(() => {
+    if (!!state.animatedMarble && !!state.moveMarbleReference) {
+      moveMarble(
+        state.moveMarbleReference.gameId,
+        state.moveMarbleReference.coordinates,
+        state.moveMarbleReference.playerForAPI,
+        state.moveMarbleReference.direction,
+        state.moveMarbleReference.token,
+      );
```
Comment (mchaffotte) [inline src/GameState/index.tsx]: It seems that your moveMarble function should have a single parameter -> moveMarbleReference?

File: src/GameState/index.tsx [hunk 4]
```diff
@@ -121,6 +147,18 @@ export default function GameState({ navigation, route, player }: any) {
...
+              dispatch({
+                type: 'animatedMarble',
+                value: {
+                  marblesCoordinate: messageParsed.event.data.marblesCoordinate,
+                  direction: messageParsed.event.data.direction,
+                },
+              });
+            }
+            break;
+
```
Comment (mchaffotte) [inline src/GameState/index.tsx:160]: It seems there is to much logic in cases of a switch

**Review by mchaffotte — CHANGES_REQUESTED** [review]

File: src/GameState/index.tsx (hunk 4 above)
Comment (arimet) [reply src/GameState/index.tsx:160]: I agree. But for the sake of time I wouldn't change the switch for now.

**Pull request merged**